lib/db/memory.js

```diff
@@ -265,36 +265,42 @@ MemoryStore.prototype = {
     for (var i = 0; i < ids.length; i++) {
       var id = ids[i];
       if (this.tokens[id].userId.toString('hex') === uid) {
-        activeClientIds.push(this.tokens[id].clientId);
+        activeClientIds.push({clientid: this.tokens[id].clientId, clientTokenTime: this.tokens[id].createdAt});
       }
     }
 
     // unique clients
-    activeClientIds.forEach(function (clientIdBuf) {
-      var clientIdHex = unbuf(clientIdBuf);
-      if (! findByBufProp(activeClients, 'id', clientIdHex)) {
-        var c = self.clients[clientIdHex];
+    activeClientIds.forEach(function (clientBuf) {
+      var clientIdHex = unbuf(clientBuf.clientid);
+      var clientTokenTime = unbuf(clientBuf.clientTokenTime);
+      var index = findByBufProp(activeClients, 'id', clientIdHex);
+      var c = self.clients[clientIdHex];
+      if (! Number.isInteger(index)) {
         if (c.canGrant === false) {
-          activeClients.push(c);
+          activeClients.push({id: c, tokenTime: clientTokenTime});
+        }
+      } else {
+        if (clientTokenTime > activeClients[index].tokenTime) {
+          activeClients[index] = {id: c, tokenTime: clientTokenTime};
```

Comment:
Ah I'm noticing we already had `c` here, let's fix that up so it is easier to follow in the future 👍